applications/DamApplication/custom_constitutive/thermal_linear_elastic_3D_law.cpp

```diff
@@ -107,13 +97,35 @@ void ThermalLinearElastic3DLaw::CalculateMaterialResponseKirchhoff (Parameters&
         this->CalculateThermalStrain(StrainVector,ElasticVariables,Temperature);
 
         StressVector = prod(ConstitutiveMatrix,StrainVector);
-    }
-    else if(Options.Is( ConstitutiveLaw::COMPUTE_STRAIN ) ) //This should be COMPUTE_THERMAL_STRAIN
-    {
+      }
+      else{
+
+	this->CalculateLinearElasticMatrix( ConstitutiveMatrix, YoungModulus, PoissonCoefficient );
+
         double Temperature;
         this->CalculateDomainTemperature( ElasticVariables, Temperature);
+
+        Vector ThermalStrainVector;
+        this->CalculateThermalStrain(ThermalStrainVector,ElasticVariables,Temperature);
+
+        StressVector = prod(ConstitutiveMatrix,(StrainVector - ThermalStrainVector));
```

Comment:
also here my advice is to generate a tmp explicitly for the difference of vectors

    array_1d<double,6> tmp = StrainVector - ThermalStrainVector
    noalias(StressVector) = prod(ConstitutiveMAtrix, tmp);

according to the ublas documentation, in this context creating explicitly the tmp is better than working with the expression resulting of StrainVector - ThermalStrainVector